src/event_handler.py

```diff
@@ -0,0 +1,44 @@
+import boto3
+import json
+import logging
+import os
+
+from base64 import b64decode
+from urllib.request import Request, urlopen
+from urllib.error import URLError, HTTPError
+
+
+# The base-64 encoded, encrypted key (CiphertextBlob) stored in the kmsEncryptedHookUrl environment variable
+ENCRYPTED_HOOK_URL = os.environ['kmsEncryptedHookUrl']
+SLACK_CHANNEL = os.environ['slackChannel']
+
+HOOK_URL = "https://" + boto3.client('kms').decrypt(CiphertextBlob=b64decode(ENCRYPTED_HOOK_URL))['Plaintext'].decode('utf-8')
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def notify_slack(event, context):
+    logger.info(f"Event: {str(event)}")
+    message = json.loads(event['Records'][0]['Sns']['Message'])
+    logger.info(f"Message: {str(message)}")
+
+    alarm_name = message['AlarmName']
+    #old_state = message['OldStateValue']
+    new_state = message['NewStateValue']
+    reason = message['NewStateReason']
+
+    slack_message = {
+        'channel': SLACK_CHANNEL,
+        'text': f"{alarm_name} state is now {new_state}: {reason}"
+    }
+
+    req = Request(HOOK_URL, json.dumps(slack_message).encode('utf-8'))
+    try:
+        response = urlopen(req)
+        response.read()
+        logger.info(f"Message posted to {slack_message['channel']}")
+    except HTTPError as e:
+        logger.error(f"Request failed: {e.code} {e.reason}")
+    except URLError as e:
+        logger.error(f"Server connection failed: {e.reason}")
```

Comment:
I guess it probably needs to be a new PR, but is it worth adding some kind of retry on failure, in case of network problems?

Reply:
We think that AWS takes care of that for us. We think

Reply:
🤞